File: Creational/Lab6/Python/apartment_builder.py
from estate_builder import *
from apartment import Apartment

class ApartmentBuilder(EstateBuilder):
    def reset(self):
        self._estate = Apartment()

    def set_rooms_number(self, number):
        self._estate.rooms_number = number
        return self

    def set_heating_type(self, heating_type):
        self._estate.heating_type = heating_type
        return self

File: Behavioral/Lab16/Typescript/src/Builders/apartment_builder.ts
import { EstateBuilder } from './estate_builder';
import { Apartment } from '../Entities/apartment';

export class ApartmentBuilder extends EstateBuilder {
    public reset(): void {
        this._estate = new Apartment();
    }

    public setRoomsNumber(number: number): ApartmentBuilder {
        const a = this._estate as Apartment;
        a.roomsNumber = number;
        return this;
    }

    public setHeatingType(type: string): ApartmentBuilder {
        const a = this._estate as Apartment;
        a.heatingType = type;
        return this;
    }
}
